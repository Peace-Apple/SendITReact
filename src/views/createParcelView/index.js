import React, { Component } from "react";
import { connect } from "react-redux";
import Header from "../../components/header"
import ParcelForm from "../../components/parcels/parcelForm";
import parcelAction from "../../actions/parcelActions";
import { toast } from "react-toastify";
import CircularProgressLoader from "../../loader/loader";

export class ParcelView extends Component {
    state = {
        receivers_name: "",
        pickup_location: "",
        destination: "",
        weight: "",
        errors: {},
        loader: {
          loading: false,
        },
      };

      componentWillReceiveProps(nextProps){
          if (nextProps.errors.error_message === "Please use character strings") {
            toast.error("Please use character strings, not numbers", {
                position: toast.POSITION.TOP_CENTER,
                autoClose: 3000,
                hideProgressBar: false,
                pauseOnHover: true,
              });
          } else {
            toast.success("You have successfully posted a parcel delivery order", {
                position: toast.POSITION.TOP_CENTER,
                autoClose: 3000,
                hideProgressBar: false,
                pauseOnHover: true,
              });
              this.props.history.push("/parcels");
          }
          this.setState({ loader: { loading: false } });
      }

      onChange = (e) => {
        this.setState({ [e.target.name]: e.target.value }); 
      }

      handleSubmit = (e) => {
        e.preventDefault();
        const parcelData = {  
            receivers_name: this.state.receivers_name,
            pickup_location: this.state.pickup_location,
            destination: this.state.destination,
            weight: this.state.weight,
        }
        this.props.parcelAction(parcelData);
        this.setState({ loader: { loading: true } });
      }

    render() {
      const loader = this.state.loader;
      return (
        <div>
          <Header />
          <CircularProgressLoader {...loader} />
          <ParcelForm
          onSubmit={this.handleSubmit}
          onChange={this.onChange}
          />
        </div>
      );
    }
}

export const mapStateToProps = state => ({
  isSuccessful: state.parcelReducer.isSuccessful,
  errors: state.parcelReducer.errors,
  message: state.parcelReducer.message,
  data: state.parcelReducer.data,
});

export default connect(mapStateToProps, 
  { parcelAction })(ParcelView);
