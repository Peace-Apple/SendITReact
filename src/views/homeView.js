import React from 'react';
import Header from '../components/header';

const HomeView = () => (
  <div>
    <Header />
  </div>
);

export default HomeView;
