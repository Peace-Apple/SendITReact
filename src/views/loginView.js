import React from "react";
import Login from "../components/auth/login/loginComponent";

const LoginView = () => (
  <div>
    <Login />
  </div>
);

export default LoginView;
