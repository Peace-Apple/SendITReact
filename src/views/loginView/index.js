import React, { Component } from "react";
import { connect } from "react-redux";
import LoginForm from "../../components/auth/loginForm";
import loginAction from "../../actions/loginActions";
import { toast } from "react-toastify";
import CircularProgressLoader from "../../loader/loader";

export class LoginView extends Component {
    state = {
        user_name: "",
        password: "",
        errors: {},
        loader: {
          loading: false,
        },
      };

      componentWillReceiveProps(nextProps){
        if(nextProps.errors.error_message === "User does not exist."){
          toast.error("A user with that email and password was not found.", {
            position: toast.POSITION.TOP_CENTER,
            autoClose: 3000,
            hideProgressBar: false,
            pauseOnHover: true,
          });
        } else { 
          toast.success("You have successfully logged in", {
            position: toast.POSITION.TOP_CENTER,
            autoClose: 3000,
            hideProgressBar: false,
            pauseOnHover: true,
            });
            this.props.history.push("/parcels");
      }
      this.setState({ loader: { loading: false } });
    }

      onChange = (e) => {
        this.setState({[e.target.name]: e.target.value});
      }

      handleSubmit = (e) => {
        e.preventDefault();
        const userData = {  
          user_name: this.state.user_name,
          password: this.state.password,
        }
        this.props.loginAction(userData);
        this.setState({ loader: { loading: true } });
      }

    render() {
      const loader = this.state.loader;
      return (
        <div>
          <CircularProgressLoader {...loader} />
          <LoginForm
          onSubmit={this.handleSubmit}
          onChange={this.onChange}
          />
        </div>
      
      );
    }
}

export const mapStateToProps = state => ({
  isSuccessful: state.loginReducer.isSuccessful,
  errors: state.loginReducer.errors,
  logged_in_as: state.loginReducer.logged_in_as,
  access_token: state.loginReducer.access_token,
});

export default connect(mapStateToProps, 
  { loginAction })(LoginView);
