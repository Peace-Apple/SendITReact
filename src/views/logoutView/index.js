import React, { Component } from "react";
import { toast, ToastContainer } from "react-toastify";

export class Logout extends Component {
  componentWillMount() {
    localStorage.removeItem("accessToken");
    localStorage.removeItem("userID");
    localStorage.removeItem("email");
    localStorage.removeItem("contact");
    localStorage.removeItem("username");
    this.props.history.push("/");
    toast.success("You have successfully Logged out", {
      position: toast.POSITION.TOP_CENTER,
      autoClose: 3000,
      hideProgressBar: false,
      pauseOnHover: true,
    });
  }

  render() {
    return (
      <div>
        <ToastContainer />
      </div>
    );
  }
}

export default Logout;
