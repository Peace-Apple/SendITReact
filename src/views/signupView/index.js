import React, { Component } from "react";
import { connect } from "react-redux";
import SignupForm from "../../components/auth/signupForm";
import signupAction from "../../actions/signupActions";
import { toast } from "react-toastify";

export class SignupView extends Component {
    state = {
        user_name: "",
        email: "",
        phone_number: "",
        password: "",
        confirmPassword: "",
        errors: {}
      };

      componentWillReceiveProps(nextProps){
        if(nextProps.errors.error_message === "email already exists"){
          toast.error("Email already exists, please use another", {
            position: toast.POSITION.TOP_CENTER,
            autoClose: 3000,
            hideProgressBar: false,
            pauseOnHover: true,
          });
        } else if(nextProps.errors.error_message === "Username already taken") { 
          toast.error("Username already taken, please use another", {
            position: toast.POSITION.TOP_CENTER,
            autoClose: 3000,
            hideProgressBar: false,
            pauseOnHover: true,
            });
        } else {
          toast.success("Your account has been created successfully, please log in", {
            position: toast.POSITION.TOP_CENTER,
            autoClose: 3000,
            hideProgressBar: false,
            pauseOnHover: true,
            });
          this.props.history.push("/login");
        }
        
      }

      handleEmailChange = (e) => {
        this.setState({email: e.target.value});
        const email = this.state.email;
        if(/^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{1,3}$/.test(email)){
          this.setState({
            errors: {}
          });
        } else {
          this.setState({
            errors: {
              email: "Please enter a correct email address."
            }
          });
        }
      }

      handleUsernameChange = (e) => {
        this.setState({user_name: e.target.value});
        const user_name = this.state.user_name;
        if(/^[A-Za-z\s]{2,30}$/.test(user_name)){
          this.setState({
            errors: {}
          })
        }else {
            this.setState({
              errors: {
                username: "Username should be atleast 3 characters long and consist of only letters"
              }
            })
        }
      }

      handlePhoneChange = (e) => {
        this.setState({phone_number: e.target.value});
        const phone_number = this.state.phone_number;
        if (/^[0-9]{9,12}$/.test(phone_number)) {
          this.setState({
            errors: {}
          });
        } else {
          this.setState({
            errors: {
              phone_number: "Phone number should be in the form 070******* and between 10 and 13 digits"
            }
          });
        }
      }
      
      handlePasswordChange = (e) => {
        this.setState({password: e.target.value});
        const password = this.state.password;
        if (password.length >= 7){
          this.setState({
            errors: {}
          });
        } else {
          this.setState({
            errors: {
              password: "Password should be atleast 8 characters long"
            }
          });
        }
      }

      handleConfirmPasswordChange = (e) => {
        const confirmPassword = e.target.value;
        const {password} = this.state;
        if (password === confirmPassword){
          this.setState({
            errors: {}
          });
        } else {
          this.setState({
            errors: {
              confirmPassword: "Passwords do not match!"
            }
          });
        }
      }

      handleSubmit = (e) => {
        e.preventDefault();
        const userData = {  
          user_name: this.state.user_name,
          email: this.state.email,
          phone_number: this.state.phone_number,
          password: this.state.password,
        }
        this.props.signupAction(userData);
      }

    render() {
      return (
        <div>
          <SignupForm
            onSubmit={this.handleSubmit}
            onChange={this.onChange}
            handleUsernameChange={this.handleUsernameChange}
            handleEmailChange={this.handleEmailChange}
            handlePasswordChange={this.handlePasswordChange}
            handleConfirmPasswordChange={this.handleConfirmPasswordChange}
            handlePhoneChange={this.handlePhoneChange}
            errors={this.state.errors}
          />
        </div>
      
      );
    }
}

export const mapStateToProps = state => ({
  isSuccessful: state.signupReducer.isSuccessful,
  errors: state.signupReducer.errors,
  message: state.signupReducer.message,
});

export default connect(mapStateToProps, 
  { signupAction })(SignupView);
