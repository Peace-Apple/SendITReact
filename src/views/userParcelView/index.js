import React, { Component } from "react";
import { connect } from "react-redux";
import Header from "../../components/header"
import userParcelsAction from "../../actions/userParcelActions";
import image from '../../assets/images/avatar.jpg';

export class UserParcelView extends Component {
    componentDidMount = () => {
         this.props.userParcelsAction(localStorage.getItem("userID"));
    }

    render() {
    const user_parcels = Object.values(this.props.parcels); 

      return (
        <div>
          <Header />
          <div className="main-content">
                <h2>Welcome to SEND-IT! We deliver as fast as you create your orders!</h2>

                <div className="parcel-link"><a href="/order">Send a Parcel</a></div>
                <div className="user_profile" id="my_profile">
                    <img src={image} alt="text"/>
                    <div>
                        <h2>{localStorage.getItem('username')}</h2>
                        <p><b>Email:</b> {localStorage.getItem('email')}</p>
                        <p><b>Contact:</b> {localStorage.getItem('contact')}</p>
                    </div>
                </div>

                <div id="user_parcels">
                <table width="99%">
                <thead>
                    <tr id="parcels_head">
                        <th>Sender</th>
                        <th>Receiver</th>
                        <th>Pickup Location</th>
                        <th>Destination</th>
                        <th>Weight</th>
                        <th>Order Date</th>
                        <th>Delivery Status</th>
                        <th>Present Location</th>
                        <th>Edit Destination</th>
                        <th>Cancel Delivery</th>
                        <th>View Detail</th>
                    </tr>
                </thead>
                <tbody>
                
                    {user_parcels.map((parcel)=> {
                       return(
                        <tr id="parcels" key={parcel.parcel_id}>
                            <td>{parcel.delivery_status}</td>
                            <td>{parcel.receivers_name}</td>
                            <td>{parcel.pickup_location}</td>
                            <td id="destinationUpdate">{parcel.destination}</td>
                            <td>{parcel.weight}</td>
                            <td>{parcel.order_date}</td>
                            <td id="parcelCancel">{parcel.delivery_status}</td>
                            <td>{parcel.present_location}</td>
                            <td>
                            <p className="links">
                            <label><button >Edit</button></label>
                            </p>
                            </td>
                            <td>
                            <p className="links">
                            <label><button >Cancel</button></label>
                            </p>
                            </td>
                            <td><p className="links">
                                <label><a >View</a></label>
                            </p></td>
                        </tr>
                         )
                         
                      })
                    }
                 
                    </tbody>
                </table>

                </div>
            </div>
        </div>
      );
    }
}

export const mapStateToProps = state => ({
   parcels: state.userParcelsReducer.parcels,
});

export default connect(mapStateToProps, 
  { userParcelsAction })(UserParcelView);
