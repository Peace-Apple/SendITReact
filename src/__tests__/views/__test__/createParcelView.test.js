import React from 'react';
import { shallow, mount } from 'enzyme';
import { ParcelView, mapStateToProps } from '../../../views/createParcelView';

describe('create parcel view', () => {
  it('should render without crashing', () => {
    const props = {};
    const wrapper = shallow(<ParcelView {...props} />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();
  });

  it('should call handle submit', () => {
    const props = {
      parcelAction: jest.fn(),
      isSuccessful: false,
      history: {
        push: jest.fn(),
      },
    };
    const wrapper = shallow(<ParcelView {...props} />);
    const instance = wrapper.instance();
    instance.handleSubmit({ preventDefault: jest.fn() });
    expect(props.parcelAction).toHaveBeenCalled();
  });

  it('should map state to props', () => {
    const initialState = {
      parcelReducer: {
        isSuccessful: false,
        errors: {},
      },
    };
    expect(mapStateToProps(initialState)).toEqual({
      isSuccessful: false,
      errors: {},
    });
  });

  it('should call on change', () => {
    const props = {
      parcelAction: jest.fn(),
      isSuccessful: false,
      history: {
        push: jest.fn(),
      },
    };
    const wrapper = shallow(<ParcelView {...props} />);
    const instance = wrapper.instance();
    instance.onChange({ target: { name: 'receivers_name', value: 'Apple' } });
    expect(instance.state.receivers_name).toEqual('Apple');
  });

  it('should call component will receive props when there is an error message', () => {
    const props = {
      parcelAction: jest.fn(),
      isSuccessful: false,
      history: {
        push: jest.fn(),
      },
      errors: {
        error_message: 'Please use character strings',
      },
    };
    const wrapper = shallow(<ParcelView {...props} />);
    wrapper.setProps({ errors: { error_message: 'Please use character strings' } });
    expect(props.errors.error_message).toBe('Please use character strings');
  });

  it('should call component will receive props when there is no error message', () => {
    const props = {
      parcelAction: jest.fn(),
      isSuccessful: false,
      history: {
        push: jest.fn(),
      },
      errors: {
        error_message: '',
      },
    };
    const wrapper = shallow(<ParcelView {...props} />);
    wrapper.setProps({ errors: { error_message: '' } });
    expect(props.errors.error_message).toBe('');
  });
});
