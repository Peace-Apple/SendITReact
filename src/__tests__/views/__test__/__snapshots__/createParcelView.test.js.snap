// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create parcel view should render without crashing 1`] = `
ParcelView {
  "context": Object {},
  "handleSubmit": [Function],
  "onChange": [Function],
  "props": Object {},
  "refs": Object {},
  "state": Object {
    "destination": "",
    "errors": Object {},
    "loader": Object {
      "loading": false,
    },
    "pickup_location": "",
    "receivers_name": "",
    "weight": "",
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <ParcelView />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div>
        <Header />
        <CircularProgressLoader
          loading={false}
        />
        <ParcelForm
          onChange={[Function]}
          onSubmit={[Function]}
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
